torchci/rockset/commons/__sql/weekly_force_merge_stats.sql

```diff
@@ -0,0 +1,174 @@
+-- gets percentage of total force merges, force merges with failures, and force merges without failures (impatient)
+WITH
+    issue_comments AS(
+        SELECT
+            issue_comment.user.login,
+            issue_comment.author_association,
+            issue_comment.body,
+            issue_comment.issue_url,
+            issue_comment.html_url,
+            issue_comment.created_at,
+            issue_comment._event_time,
+            CAST(
+                SUBSTR(
+                    issue_comment.issue_url,
+                    LENGTH(
+                        'https://api.github.com/repos/pytorch/pytorch/issues/'
+                    ) + 1
+                ) as INT
+            ) as pr_num
+        FROM
+            commons.issue_comment
+        WHERE
+            (
+                issue_comment.body LIKE '%pytorchbot merge%'
+                OR issue_comment.body LIKE '%pytorchmergebot merge%'
+            ) -- AND _event_time >= PARSE_DATETIME_ISO8601(:startTime)
+            -- AND _event_time < PARSE_DATETIME_ISO8601(:stopTime)
+            AND issue_comment.user.login NOT LIKE '%pytorch-bot%'
+            AND issue_comment.user.login NOT LIKE '%facebook-github-bot%'
+            AND issue_comment.user.login NOT LIKE '%pytorchmergebot%'
+            AND issue_comment.issue_url LIKE '%https://api.github.com/repos/pytorch/pytorch/issues/%'
+    ),
+    all_merges AS (
+        SELECT
+            DISTINCT m.skip_mandatory_checks,
+            LENGTH(m.failed_checks) AS failed_checks_count,
+            m.ignore_current,
+            m.is_failed,
+            m.pr_num,
+            m.merge_commit_sha,
+            max(c._event_time) as time,
+        FROM
+            commons.merges m
+            inner join issue_comments c on m.pr_num = c.pr_num
+        WHERE
+            m.owner = 'pytorch'
+            AND m.project = 'pytorch'
+            AND m.merge_commit_sha != '' -- only consider successful merges
+            AND m._event_time >= PARSE_DATETIME_ISO8601(:startTime)
+            AND m._event_time < PARSE_DATETIME_ISO8601(:stopTime) -- AND m.pr_num in 
+        GROUP BY
+            m.skip_mandatory_checks,
+            m.failed_checks,
+            m.ignore_current,
+            m.is_failed,
+            m.pr_num,
+            m.merge_commit_sha -- and m.pr_num = 104137
+    ),
+    force_merges_with_failed_checks AS (
+        SELECT
+            IF(
+                (skip_mandatory_checks = true)
```

Comment:
This condition doesn't look right when I compare it with https://console.rockset.com/lambdas/details/metrics.force_merge_red_percent.  It should be I think:

```
      (
        skip_mandatory_checks = true
        AND failed_checks_count > 0
      )
```

`skip_mandatory_checks` means `-f` is used, and `failed_checks_count` tells you the number of failures (minus broken trunk, flaky, and unstable) at the time of the merge.